### ISSUE TITLE
Problem uploading with new media browser when hosting site in virtual directory

### ISSUE THREAD

**Issue by libardo** [n2cms/n2cms#683]
See #675

**Comment by SntsDev** [comment]
@laurachan, 

Just to be clear. If you add a *virtual directory* as an upload folder the Media Browser works.
But your problem is not using a virtual directory, but having the whole N2 Site under an *APPLICATION*, right? (which in the Url looks like a dir).

With virtual "upload" directories it works if you do it in this way:
```
<n2 xmlns="http://n2cms.com/schemas/configuration/v3">
  ...
  <edit>
    ...
    <uploadFolders>
        <add path="~/mediaPR" title="MediaPR" />
    </uploadFolders>
  </edit>
</n2>
```
Where "mediaPR" is a virtual directory defined in IIS.

BUT, as I mentioned, I think that your problem is when you use an *Application*. Please try o be clear here. Also take a look at the console to see the URL to the MediaBrowser handler. Here is the problem (with this setup) because it requests "selected=/AppFolder/Upload/" instead of only request "/upload".

So I'm going to fix this, but please tell me if this is your problem.
Best

**Comment by libardo** [comment]
I reproduces this by hosting the whole site in an application below a site.

**Comment by SntsDev** [comment]
Exactly, the problem arises when it is as an Application below a site. @laurachan, do you agree?

**Comment by SntsDev** [comment]
I have another question regarding this:

Why are you using an App? Just because the web has to be BELOW a site? Or is it for saving hosting costs? (you create an App and then the domain is served by it, as in the case of Godaddy).

The distinction is important because in a normal App (served by a base domain + path-to-app) the URLs of the images should be "/path-to-app/upload/my-image.png". On the other hand, if using the App points to another domain and this is the "root", then the images paths must NOT contain the "/path-to-app/" part.

My guess is that it must be the whole path (including "/path-to-app") and in your views & helpers you should omit the path-to-app if needed.

What's your opinion @libardo?

**Comment by laurachan** [comment]
Hi @SntsDev, 

I checked with @ahmadh21 about our group's IIS setup. Yes, we do have our N2CMS project set up as an application, whose virtual path on Local IIS would be something like localhost/MyVirtualDir. The setup is meant to mimic our production environment, which has to host a few of our other ASP.NET web application projects as well. 

The upload directory path would be localhost/MyVirtualDir/upload in this setup. I was able to obtain the following information about the path to MediaBrowser.ashx from inspecting the media browser's markup with the Chrome console. Let me know if there is another place you would like me to check.

![image](https://cloud.githubusercontent.com/assets/3029296/11129196/ee2c7fae-8932-11e5-988e-56b40702ff6b.png)

Anyways, thanks for the update in #686; I will try upgrading to N2CMS 2.9.1 as soon as possible.
